docs/design/loadpath.md

```diff
@@ -0,0 +1,164 @@
+perform this operation.
+
+While this is still a work in progress, it is a promising development and
+something we can leverage to achieve our goals.
+
+The integration would look something like this:
+
+1. Using the bpfd CSI plugin in Kubernetes, it would be possible to express -
+   through the volume attributes - that a pod should be given a token with
+   whichever permissions are required and that it should be placed to a given
+   location on the bpf filesystem.
+1. At this point, there is nothing more for bpfd to do, since the token is
+   already in place and the pod can load eBPF programs as it sees fit.
+
+This would also be exposed through a `bpfd token create` command, which would
+create a token with the given permissions and place it at the given location
+on the bpf filesystem. This could then be mounted into a container and used
+to load eBPF programs.
+
+The drawback to this approach is that it only grants permissions to
+make bpf() syscalls without capabilities. There are still some capabilities
+that might be required - i.e IPC_LOCK or PERFMON. It's unclear at this point
+whether this will run foul of the kernels capabilities checks - whether
+it's checking for capabilities in the current userns or the parent userns.
+
+Not only that, but it will never support some eBPF tracing use-cases
+i.e. attaching a uprobe to a binary in another pid/mount namespace. Therfore,
+we may still need a "rootful" eBPF program loader for these use-cases.
+
+[BPF Token patchset]: https://lore.kernel.org/bpf/CAEiveUeDLr00SjyU=SMSc4XbHSA6LTn4U2DHr12760rbo5WqSw@mail.gmail.com/T/
+
+### Syscall Proxying (seccomp-notify)
+
+seccomp-notify is another possible way to solve the problem.
+As described in a blog post titled [The Seccomp Notifier: New Frontiers in Unprivileged Container Development].
+
+The premise here is that syscalls are proxied into user-space, where they can
+be handled by a user-space process that has the required capabilities, and then
+the result of the syscall is returned to the caller.
+
+When presented as an alternative to the BPF Token patchset, it was dismissed
+since Christian Brauner (the author of the blog post) described it as
+"impractical". However, there is a project called [Seitan] which seeks to
+address some of the practicality issues, and also provide container runtime
+integration.
+
+The integration would look something like this:
+
+1. bpfd might be able to provide information to Seitan that can be used to
+   generate the seccomp filters that are required to proxy syscalls.
+1. bpfd might do nothing, except for preparing a bpffs filesystem for the
+   container runtime to use.
+
+[The Seccomp Notifier: New Frontiers in Unprivileged Container Development]: https://people.kernel.org/brauner/the-seccomp-notifier-new-frontiers-in-unprivileged-container-development
+[Seitan](https://seitan.rocks/seitan/about/)
+
+### Authoritative Policy (via LSM or otherwise)
```

Comment:
We should probably add a note on the approach from 

 http://vger.kernel.org/bpfconf2023_material/Lorenz_Bauer_-_BPF_signing_using_fsverity_and_LSM_gatekeeper.pdf 

and 

https://github.com/isovalent/bpf-verity